src/operations/transforms/mirror.js

```diff
@@ -0,0 +1,53 @@
+const flatten = require('../utils/flatten')
+
+const mat4 = require('../../math/mat4')
+const plane = require('../../math/plane')
+
+const {geom2, geom3, path2} = require('../../geometry')
+
+/**
+ * Mirror the given object(s) using the given options (if any)
+ * Note: The normal should be given as 90 degrees from the plane origin.
+ * @param {Object} options - options for mirror
+ * @param {Array} options.origin=[0,0,0] - the origin of the plane
+ * @param {Array} options.normal=[0,0,1] - the normal vector of the plane
+ * @param {Object|Array} objects - the objects(s) to mirror
+ * @return {Object|Array} the mirrored object(s)
+ *
+ * @example
+ * let newsphere = mirror({normal: [0,0,10]}, cube({center: [0,0,15], radius: [20, 25, 5]}))
+ */
+const mirror = (options, ...objects) => {
+  const defaults = {
+    origin: [0, 0, 0],
+    normal: [0, 0, 1] // Z axis
+  }
+  let {origin, normal} = Object.assign({}, defaults, options)
+
+  objects = flatten(objects)
+  if (objects.length === 0) throw new Error('wrong number of arguments')
+
+  const planeOfMirror = plane.fromNormalAndPoint(normal, origin)
+  const matrix = mat4.mirrorByPlane(planeOfMirror)
+
+  const results = objects.map((object) => {
+    if (path2.isA(object)) return path2.transform(matrix, object)
```

Comment:
very nice use of the new functionalities ! 
However 
- it seems a bit problematic to have hard coded checks for all the data types that we have ? this means a need to change code when adding new ones (that also includes user created types): but not sure if it will really be an issue
- I am less sure about the mirroring of paths though (trying to find a good visual representation)

Reply:
>     * it seems a bit problematic to have hard coded checks for all the data types

I'm not a big fan either, but took the easiest route.

Definitely, we have to determine if an object is the correct type, so a function call is required to interrogate the object.

We could use some kind of pre-defined jump table to the geometry specific function, but that also means maintenance.